### DISCUSSION TITLE
Trying to load exisitng persist database

### DISCUSSION THREAD

**Discussion by clearsitedesigns** [PromtEngineer/localGPT#467] [Q&A]
Hi all, I spent a lot of time on this the other day. I'm building some data viz on my chroma DB, but I'm trying to write another script that accesses the vector DB, but whenever I try it fails. I've tried so many things. So any help would be much appreciated!

I've tried all these things. I'm using a local LLM llama2 as the go between, and the instruct model.

https://docs.trychroma.com/embeddings

**Comment by maxng07** [comment]
Hi, when you say failed, what do you get?
I have done similar, however instead of using LocalGPT, I'm using PrivateGPT as I do not have GPU. Few things to take note, you have to use back the same embeddings model you used to ingest the data into Chroma for your query. ChromaDB default uses `all-MiniLM-L6-v2`.

I was curious how the tokens looks like and what is the distance of the return response. These are not exposed on PrivateGPT and the langchain scripts. Hence I used the ChromaDB to have a peek. It took me a few days to realise, it could be an issue with the indexing. Using PrivateGPT script, I could get a response when I perform the query, however, if I use the Chroma embeddings script shown in the same URL I yielded no result. I couldn't recreate the index, hence I re-ingest 2 documents. And run my chroma py script again and it produce the results I wanted. I'm not sure if this is the same case as you, but it is highly likely the chroma index is corrupted or not able to be fetched by ChromaDB. After I re-ingest, it fixed the issues. It dumps out the matches up to the n results you configure. I did not pass it to LLM. 
```
This is my Chroma script I used, they are reference from Chroma documentation
import chromadb
from chromadb.config import Settings
from chromadb.utils import embedding_functions
from chromadb import utils

from constants import CHROMA_SETTINGS

query = "xxxxxx"

embeddings = "all-MiniLM-L6-v2"
st_embed = embedding_functions.SentenceTransformerEmbeddingFunction(model_name=embeddings)
chroma_client = chromadb.Client(Settings(
                            chroma_db_impl="duckdb+parquet",
                            persist_directory="./db"))


default_ef = embedding_functions.DefaultEmbeddingFunction()
print(chroma_client.list_collections())

collection = chroma_client.get_collection(name="langchain", embedding_function=st_embed)
print(collection.count())

dbresults = collection.query(
              query_texts=[query],
              n_results=10,
              include=["embeddings","documents", "metadatas", "distance"   
          )

print(dbresults)
print(collection)

```

**Reply by clearsitedesigns** [reply]
Let's me give it a try I have to re-work a lot of my code after the changes.
